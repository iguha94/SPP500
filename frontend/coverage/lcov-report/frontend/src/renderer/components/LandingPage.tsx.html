<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for frontend/src/renderer/components/LandingPage.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">frontend/src/renderer/components</a> LandingPage.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import '../css/landing_page.css';
&nbsp;
export const LandingPage: React.StatelessComponent&lt;{}&gt; = <span class="fstat-no" title="function not covered" >() </span>=&gt; {
<span class="cstat-no" title="statement not covered" >	return (</span>
		&lt;div className="landing-container"&gt;
			&lt;img src={require('../../../../doc/art/DM-Tools-Logo.png')} className="logo"&gt;&lt;/img&gt;
			&lt;h1&gt;DM Tools&lt;/h1&gt;
			&lt;h3&gt;One-stop platform for Dungeon Masters!&lt;/h3&gt;
		&lt;/div&gt;
	);
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 15 2019 16:15:32 GMT-0600 (Central Standard Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
