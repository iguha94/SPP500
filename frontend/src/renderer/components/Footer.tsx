import React from 'react';

import '../css/footer.css';

export const Footer: React.StatelessComponent<{}> = () => {
	return (
		<div className="footer-container">
			<h5>Â© D&D Tools 2019 - All rights reserved</h5>
		</div>
	);
}
