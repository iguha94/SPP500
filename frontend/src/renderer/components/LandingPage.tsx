import React from 'react';
import '../css/landing_page.css';

export const LandingPage: React.StatelessComponent<{}> = () => {
	return (
		<div className="landing-container">
			<img src={require('../../../../doc/art/DM-Tools-Logo.png')} className="logo"></img>
			<h3>One-stop platform for Dungeon Masters!</h3>
		</div>
	);
}
