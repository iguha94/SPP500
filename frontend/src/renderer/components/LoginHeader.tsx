import * as React from 'react';
import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';
import Snackbar from '@material-ui/core/Snackbar';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

import '../css/login_header.css';

type AppProps = {}

export class LoginHeader extends React.Component<{}> {
	constructor(props: AppProps) {
		super(props);
		this.state = {
			user: {
				// Todo: Accept e-mail address as username (i.e. as a way to login)
				username: "",
				password: ""
			},

			snackbar: {
				open: false,
				message: ""
			}
		}
	}

	handleUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		const user = this.state.user
		this.setState({
			user: { ...user, username: event.target.value }
		})
	}

	handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		const user = this.state.user
		this.setState({
			user: { ...user, password: event.target.value }
		})
	}

	openSnackbar = (messageText: string) => {
		const snackbar = this.state.snackbar
		this.setState({ snackbar: {...snackbar, open: true, message: messageText }});
	};

	closeSnackbar = () => {
		const snackbar = this.state.snackbar
		this.setState({ snackbar: {...snackbar, open: false }});
	};

	render() {
		const login = (event: React.FormEvent) => {
			event.preventDefault();
			this.closeSnackbar();
			this.requestLogin();
		}

		return (
			<div className="login-header-container">
				<div className="left-block">
					<img src={require('../../../../doc/art/DM-Tools-Logo.png')} className="logo"></img>
				</div>
				<div className="right-block">
				<form onSubmit={login}>
					<Input className="input" id="username" placeholder="Username" name="username" autoComplete="username" value={this.state.user.username} onChange={this.handleUsernameChange} autoFocus required/>
					<Input className="input" id="password" placeholder="Password" name="password" type="password" autoComplete="current-password" value={this.state.user.password} onChange={this.handlePasswordChange} required/>
					<Button className="button" variant="contained" color="primary" type="submit">Login</Button>
				</form>
				</div>
					<Snackbar
						open={this.state.snackbar.open}
						message={<span id="message-id">{this.state.snackbar.message}</span>}
						anchorOrigin={{
							vertical: 'bottom',
							horizontal: 'left'
						}}
						action={[
							<IconButton
								key="close"
								aria-label="Close"
								color="inherit"
								onClick={this.closeSnackbar}
							>
								<CloseIcon/>
							</IconButton>
						]}
					/>
			</div>
		);
	}


	requestLogin() {
		var context = this;
		var request = require("request");
		var options = { method: 'POST',
			url: 'http://3.17.205.229:3000/login',
			headers:
			{
				'Cache-Control': 'no-cache',
				'Content-Type': 'application/json'
			},
			body:
			{
				username: this.state.user.username,
				password: this.state.user.password
			},
			json: true
		};

		request(options, function (error:string, response:string, body:string) {
			var message = response.body.message;
			var token = response.body.token;
			context.openSnackbar(message);
			
			if (error) throw new Error(error);
		});
	}
}
