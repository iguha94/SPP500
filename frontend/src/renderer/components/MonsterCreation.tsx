import React from 'react';
import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';

import "../css/create_monster.css"

const validateForm = (event: React.FormEvent) => {
	//console.log("Hello!")
	event.preventDefault();
}

export const MonsterCreation: React.StatelessComponent<{}> = () => {
	return (
		<div className="monster-creation-container">
			<form onSubmit={validateForm} >
				<h1 className="page-title">Create a Monster</h1>
				<div className="form-group">
					<TextField autoFocus margin="dense" id="name" label="Email Address" helperText="Test" fullWidth required/>
				</div>
				<Button className="button" variant="contained" color="primary" type="submit"> Create Monster </Button>
			</form>
		</div>
	);
}