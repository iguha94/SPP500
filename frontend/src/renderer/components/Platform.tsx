import * as React from 'react';

type AppProps = {}

export class Platform extends React.Component<{}> {
	constructor(props: AppProps) {
		super(props);
	}

	render() {
		return (
			<div className="platform-container">
				You are logged in, welcome to the D&D Tools!
			</div>
		);
	}
}
