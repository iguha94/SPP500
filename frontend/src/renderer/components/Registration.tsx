import * as React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import '../css/registration.css';

type AppProps = {}

export class Registration extends React.Component<{}> {
	constructor(props: AppProps) {
		super(props);
	}
	render() {
		return (
			<div className="registration-container">
				<React.Fragment>
					<Typography variant="h6" gutterBottom>
						Register Now!
					</Typography>
					<Grid container spacing={24}>
						<Grid item xs={12} sm={6}>
							<TextField
								required
								id="firstName"
								name="firstName"
								label="First name"
								fullWidth
								autoComplete="fname"
							/>
						</Grid>
						<Grid item xs={12} sm={6}>
							<TextField
								required
								id="lastName"
								name="lastName"
								label="Last name"
								fullWidth
								autoComplete="lname"
							/>
						</Grid>
						<Grid item xs={12}>
							<TextField
								required
								id="email"
								name="email"
								label="E-mail Address"
								type="email"
								fullWidth
								autoComplete="email"
							/>
						</Grid>
						<Grid item xs={12} sm={6}>
							<TextField
								required
								id="username"
								name="username"
								label="Username"
								fullWidth
								autoComplete="username"
							/>
						</Grid>
						<Grid item xs={12} sm={6}>
							<TextField 
								required
								id="password"
								name="password"
								label="Password"
								type="password"
								fullWidth
							/>
						</Grid>
						<Grid item xs={12}>
							{/*Needed for spacing purposes - No functionality*/}
						</Grid>
					</Grid>
					<Button className="button" variant="contained" color="primary">Register</Button>
				</React.Fragment>
			</div>
		);
	}
}
