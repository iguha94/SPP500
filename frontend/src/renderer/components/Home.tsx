import React from 'react';
import {LoginHeader} from './LoginHeader';
import {Footer} from './Footer';
import {LandingPage} from './LandingPage';
import {Registration} from './Registration';
import '../css/app.css';


export const HomePage: React.StatelessComponent<{}> = () => {
	return (
		<div className="container-fluid">
			<div className="header"><LoginHeader/></div>
				<div className="horizontal-block">
					<div className="align-left"><LandingPage/></div>
					<div className="align-right"><Registration/></div>
				</div>
			<div className="footer"><Footer/></div>
		</div>
	);
}

