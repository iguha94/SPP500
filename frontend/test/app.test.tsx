import * as React from 'react';
import {shallow} from 'enzyme';
import {HomePage} from '../src/renderer/components/Home';

jest.mock('../src/cookie');

describe('Test the Landing/Registration/Login Page', () => {
	const appInstance = shallow(<HomePage/>);

	it('renders without crashing', () => {
		expect(appInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(appInstance).toMatchSnapshot();
	});
});
