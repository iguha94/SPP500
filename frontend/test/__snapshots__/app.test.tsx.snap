// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Landing/Registration/Login Page renders correctly when the page is loaded 1`] = `
<BrowserRouter>
  <div>
    <ul>
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
        <Link
          replace={false}
          to="/registration"
        >
          Home
        </Link>
      </li>
    </ul>
    <hr />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/registration"
    />
  </div>
</BrowserRouter>
`;
